lib/semantics/resolve-names-utils.h

```diff
@@ -64,6 +68,12 @@ class GenericSpecInfo {
   void Analyze(const parser::GenericSpec &);
 };
 
+// Analyze a parser::ArraySpec or parser::CoarraySpec into the provide ArraySpec
```

Comment:
"provided"

But why not return an `ArraySpec` as a value?

Reply:
I initially had it that way but changed it because this way was slightly simpler. I'm happy to change it back if you think returning a value is better.

Reply:
It would be more clear, IMO, by removing any question about whether the result argument is meant to be updated in place, reassigned wholesale, or whatever.

Reply:
Done